googleapiclient/errors.py

```diff
@@ -44,9 +44,9 @@ def _get_reason(self):
     """Calculate the reason for the error from the response content."""
     reason = self.resp.reason
     try:
-      data = json.loads(self.content)
+      data = json.loads(self.content.decode('utf-8'))
```

Comment:
Looking at it again, I think your idea is better.  What do you think about checking for bytes and raising an error instead of using the assert?